app/Services/Rogue.php

```diff
@@ -99,4 +98,23 @@ public function updatePost($postId, $input)
 
         return $post;
     }
+
+    /**
+     * Get an action from Rogue based on CallPower campaign id.
+     *
+     * @param integer $callpowerCampaignId
+     * @return array $action
+     */
+    public function getActionFromCallPowerCampaignId($callpowerCampaignId)
+    {
+        $action = $this->asClient()->get('v3/actions', [
+            'filter' => ['callpower_campaign_id' => $callpowerCampaignId],
+        ]);
+
+        if (! $action['data']) {
+            throw ValidationException::withMessages('Unable to get action data for CallPower campaign id : ' . $callpowerCampaignId);
```

Comment:
Since this is happening outside the context of a HTTP request, should this just be a plain old `Exception`?